cc/toolchains/artifacts.bzl

```diff
@@ -0,0 +1,103 @@
+# Copyright 2025 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rules to turn artifact categories into targets."""
+
+load(":cc_toolchain_info.bzl", "ArtifactCategoryInfo", "ArtifactNamePatternInfo")
+
+visibility("public")
+
+def _cc_artifact_category_impl(ctx):
```

Comment:
Can we put this rule in a seperate file and then make that file's visibility  only to this repo?

Don't think we should allow users to create their own artifact category, rules_cc should just provide the build file you defined earlier.

Reply:
My reasoning for allowing users to define custom categories was that we allow the same for action types.

Moving it to `impl` and making private.

Reply:
Probably shouldn't allow it for action types. Pretty sure the reason was that when I first created it, I saw some action types that weren't fully covered by `action_names.bzl` (I think bazel's and rules_cc's were out of sync), but I think that's since been fixed.